src/types/instantsearch.ts

```diff
@@ -91,11 +92,26 @@ export type NumericRefinement = {
 
 export type Refinement = FacetRefinement | NumericRefinement;
 
+export type Router = {
+  onUpdate(callback: (route: string) => void): void;
+  read(): UiState;
+  write(route: string): void;
+  createURL(state: Partial<SearchParameters>): string;
+  dispose(): Widget['dispose'];
+};
+
+export type StateMapping = {
+  stateToRoute(state: UiState): string;
```

Comment:
The return value is not a `string` but an object that represents the `uiState` for the URL. 

By default it's our `UIState` but it can be different if `StateMapping` is implemented by the user.